headers/headers.go

```diff
@@ -10,16 +10,47 @@ import (
 	"github.com/ONSdigital/log.go/log"
 )
 
-var (
-	// CollectionIDKey is the name used for a collection ID http request header
-	CollectionIDKey = "Collection-Id"
+const (
+	// CollectionIDHeader is the name used for a collection ID http request header
+	CollectionIDHeader = "Collection-Id"
+
+	// UserAuthTokenHeader is the user Florence auth token header name
+	UserAuthTokenHeader = "X-Florence-Token"
```

Comment:
Opinionated view: I think from now on this header should have a single name and all references to it should use that same name. This should help avoid confusion. I've gone with `UserAuthTokenHeader` but happy to argue over a different name.

Reply:
Again just my personal view, I think `UserAuthToken` would suffice as these are all headers. I think as the package name is explicit and accurate for what it holds we don't need to append the names with header in this instance (so long as the content of this package doesn't ever get out of scope of itself).

I strongly agree about us setting single name and doing it now.